vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -232,6 +233,18 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     HttpServerRequest request = context.request();
     if (!methods.isEmpty() && !methods.contains(request.method())) {
+      context.response().putHeader(HttpHeaders.ALLOW, String.join(", ", () -> new Iterator<CharSequence>() {
+        final Iterator<HttpMethod> it = methods.iterator();
+        @Override
+        public boolean hasNext() {
+          return it.hasNext();
+        }
+
+        @Override
+        public CharSequence next() {
+          return it.next().name();
+        }
+      }));
```

Comment:
This is a large chunk of code that is out of context in this method.
Suggestion: Make a static class inside inside `RouteImpl` that receives the methods Set and returns a function. That function creates and returns the iterator.
Create a new field + methods on `RoutingContext` (or related) to hold the default error warning